subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_rest

```diff
@@ -10,17 +10,31 @@ menuconfig NRF_CLOUD_REST
 
 if NRF_CLOUD_REST
 
-config NRF_CLOUD_REST_HOST_NAME
-	string "nRF Cloud REST API hostname"
-	default "api.nrfcloud.com"
-
-config NRF_CLOUD_REST_FRAGMENT_SIZE
-	int "Fragment size for REST API downloads"
-	range 128 1700
-	default 1700
-
-module = NRF_CLOUD_REST
-module-str = nRF Cloud REST
-source "subsys/logging/Kconfig.template.log_config"
+	config NRF_CLOUD_REST_HOST_NAME
+		string "nRF Cloud REST API hostname"
+		default "api.nrfcloud.com"
+
+	config NRF_CLOUD_REST_AUTOGEN_JWT
+		bool "nRF Cloud REST Auto-generate JWTs"
+		default y
+		select MODEM_JWT
+
+	menu "JWT Autogeneration Settings"
```

Comment:
The `visible if` condition only works on menu defs, and I cannot use a `depends on` because this will cause a compiler error `CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT_LIFETIME` since I used `IS_ENABLED(CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT)` instead of a pure preprocessor directive for that branch of code. (Compiler optimization should still remove that branch if it's disabled, but in order to get that far, `CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT_LIFETIME` still has to be visible)